src/Nancy/ModelBinding/DefaultModelBinderLocator.cs

```diff
@@ -27,7 +27,11 @@ public class DefaultModelBinderLocator : IModelBinderLocator
         public DefaultModelBinderLocator(IEnumerable<IModelBinder> binders, IBinder fallbackBinder)
         {
             this.fallbackBinder = fallbackBinder;
-            this.binders = binders;
+
+            if (binders != null)
+            {
+                this.binders = binders.ToArray();

```

Comment:
This is the same as my comment on the other issue, but I believe you can do:

`this.binders = new ReadOnlyCollection(binders);`

This means it will work on the original collection and not copy it to a new one.